.github/workflows/codespeed-agent-benchmarks.yaml

```diff
@@ -0,0 +1,169 @@
+name: Codespeed Agent Benchmark Reusable Workflow
+
+on:
+  workflow_call:
+    inputs:
+      python-version:
+        description: Python Version
+        type: string
+        required: true
+      agent_config:
+        description: "Path to the agent config file to use."
+        type: string
+        required: true
+      codespeed_executable:
+        description: "CodeSpeed executable name."
+        type: string
+        required: true
+      codespeed_environment:
+        description: "CodeSpeed environment name."
+        type: string
+        default: GitHub Hosted Action Runner
+        required: false
+      run_time:
+        description: "How long to run the capture for (in seconds)."
+        type: number
+        default: 120
+        required: false
+      capture_interval:
+        description: "How often to capture agent process level metrics during the process run time (in seconds)."
+        type: number
+        default: 5
+        required: false
+      agent_pre_run_command:
+        description: "Optional bash command / script to run before starting the agent and the metrics capture script."
+        type: string
+        default: ""
+        required: false
+      agent_post_run_command:
+        description: "Optional bash command / script to run after starting the agent and the metrics capture script."
+        type: string
+        default: ""
+        required: false
+      agent_server_host:
+        description: "Value for the server_attributes.serverHost agent configuration option."
+        type: string
+        default: "ci-codespeed-benchmarks"
+        required: false
+      capture_agent_status_metrics:
+        description: "True to capture additional metrics exposed via agent status command."
+        type: boolean
+        default: false
+        required: false
+      capture_line_counts:
+        description: "True to submit log line counts for each log level to CodeSpeed."
+        type: boolean
+        default: false
+        required: false
+      dry_run:
+        description: "True to enable dry run which runs the benchmarks without submitting data to CodeSpeed."
+        type: boolean
+        default: false
+        required: false
+
+jobs:
+  codespeed-reusable-workflow:
```

Comment:
Perhaps agent benchmark or similar would be better name for this job.

Reply:
EDIT: I see you give actual jobs which call this re-usable flow better names so it's probably fine :)

Reply:
EDIT: Looks like job names are not good so I will go ahead and try to rename them / give them better name - https://github.com/scalyr/scalyr-agent-2/actions/runs/2902530969

Reply:
Pushed a change to make it a bit better (https://github.com/scalyr/scalyr-agent-2/runs/7948166748?check_suite_focus=true), but it's still not great - not much we can do though with the current UI job name truncation.